src/Repository/DefaultUnleashRepository.php

```diff
@@ -239,19 +247,196 @@ private function parseFeatures(array $body): array
             }
 
             $featureVariants = $this->parseVariants($feature['variants'] ?? []);
+            $dependencies = $this->parseDependencies($feature['dependencies'] ?? [], $features, $hasUnresolvedDependencies);
 
-            $features[$feature['name']] = new DefaultFeature(
+            $featureDto = new DefaultFeature(
                 $feature['name'],
                 $feature['enabled'],
                 $strategies,
                 $featureVariants,
                 $feature['impressionData'] ?? false,
+                $dependencies,
             );
+            if ($hasUnresolvedDependencies) {
+                $unresolved[] = $featureDto;
+            } else {
+                $features[$feature['name']] = $featureDto;
+            }
         }
 
+        // $features are passed by reference so that the method can modify them
+        $this->handleUnresolvedDependencies($unresolved, $features);
+
         return $features;
     }
 
+    /**
+     * All unresolved dependencies are solved in this method (if possible).
```

Comment:
This is fine for now. Every SDK will be much more complex with multi-level deps. However in our initial research we haven't found a need for this extra complexity.